web/src/components/filings-calendar/tax-access-modal/TaxAccessStepTwo.test.tsx

```diff
@@ -188,6 +188,38 @@ describe("<TaxAccessStepTwo />", () => {
       });
     });
 
+    it("updates businessName on submit if tax filing is success", async () => {
+      renderComponent(userDataWithPrefilledFields);
+      mockApi.postTaxFilingsOnboarding.mockResolvedValue({
+        ...userDataWithPrefilledFields,
+        profileData: {
+          ...userDataWithPrefilledFields.profileData,
+          municipality: undefined,
+        },
+        taxFilingData: generateTaxFilingData({
+          state: "SUCCESS",
+          businessName: userDataWithPrefilledFields.profileData.businessName,
+          errorField: undefined,
+        }),
+      });
+      fireEvent.change(screen.getByLabelText("Business name"), {
+        target: { value: "zoom" },
+      });
+      fireEvent.change(screen.getByLabelText("Tax id"), {
+        target: { value: "999888777666" },
+      });
+      clickSave();
+      await waitFor(() => {
```

Comment:
Asking because I'm not sure- do these need to be separate waitFors?

Reply:
good catch, propagated from copying the test above it, cleaned up that as well.